core/tools/blueos_startup_update/boot/pi5overlays.py

```diff
@@ -0,0 +1,167 @@
+import re
+from typing import List, Optional
+
+from loguru import logger
+
+from commonwealth.utils.commands import run_command
+
+
+class Overlay:
+    def __init__(self, name: str, args: List[str], index: Optional[int] = None):
+        self.name = name
+        self.args = args
+        self.index = index
+
+    def __repr__(self) -> str:
+        joined_args = " ".join(self.args)
+        return f"{self.name} {joined_args}"
+
+    def wrongly_set_in(self, overlays: List["Overlay"]) -> bool:
+        for loaded_overlay in overlays:
+            if loaded_overlay.name == self.name and loaded_overlay.args != self.args:
+                return True
+        return False
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, Overlay):
+            return False
+        return self.name == other.name and self.args == other.args
+
+    def reload(self) -> None:
+        command_result = run_command(f"sudo dtoverlay -r {self.name} && sudo dtoverlay {self}", False)
+        logger.info(command_result)
+
+    def load(self) -> None:
+        command_result = run_command(f"sudo dtoverlay {self}", False)
+        logger.info(command_result)
+
+    @staticmethod
+    def from_string(overlay: str) -> "Overlay":
+        match = re.match(r"^((?P<index>\d+):\s+)?(?P<name>(\S)+)(?:$|\s+(?P<args>.*))", overlay)
```

Comment:
Can we extract this line into a class method so we can call it on line 127 instead of having the regex duplicated?